internal/events/aggregator.go

```diff
@@ -342,21 +393,20 @@ func (ag *aggregator) attemptMessageDispatch(ctx context.Context, msg *fftypes.M
 
 	// Validate the message data
 	valid := true
+	var customCorrelator *fftypes.UUID
 	switch {
 	case msg.Header.Type == fftypes.MessageTypeDefinition:
 		// We handle definition events in-line on the aggregator, as it would be confusing for apps to be
 		// dispatched subsequent events before we have processed the definition events they depend on.
-		msgAction, batchAction, err := ag.definitions.HandleDefinitionBroadcast(ctx, msg, data, tx)
+		state.correlator = nil
```

Comment:
😄  - has just removed the multiple returns, then this one edge case came back.

If I shift it to a return, I'd like to make it an extensible structure again like before - because this one item doesn't feel significant enough to justify its own parameter in an ever increasing list.

Reply:
I've created a bigger `HandlerResult` structure to return that can contain a `CustomCorrelator` in addition to an `Action`